crates/autopilot/src/database/auction.rs

```diff
@@ -81,20 +64,28 @@ impl QuoteStoring for Postgres {
 }
 
 impl Postgres {
-    pub async fn solvable_orders(&self, min_valid_to: u32) -> Result<SolvableOrders> {
+    pub async fn solvable_orders(
+        &self,
+        min_valid_to: u32,
+        min_surplus_fee_timestamp: DateTime<Utc>,
+    ) -> Result<SolvableOrders> {
         let _timer = super::Metrics::get()
             .database_queries
             .with_label_values(&["solvable_orders"])
             .start_timer();
 
         let mut ex = self.0.begin().await?;
-        let orders = database::orders::solvable_orders(&mut ex, min_valid_to as i64)
-            .map(|result| match result {
-                Ok(order) => full_order_into_model_order(order),
-                Err(err) => Err(anyhow::Error::from(err)),
-            })
-            .try_collect::<Vec<_>>()
-            .await?;
+        let orders = database::orders::solvable_orders(
+            &mut ex,
+            min_valid_to as i64,
+            min_surplus_fee_timestamp,
+        )
+        .map(|result| match result {
+            Ok(order) => full_order_into_model_order(order),
+            Err(err) => Err(anyhow::Error::from(err)),
+        })
+        .try_collect()
```

Comment:
Oh man... I've been using `.collect::<Result<_>>()` for so long, I didn't know this was a thing!